src/main/java/com/google/devtools/build/lib/worker/WorkerKey.java

```diff
@@ -42,6 +42,7 @@
   private final HashCode workerFilesCombinedHash;
   private final SortedMap<PathFragment, HashCode> workerFilesWithHashes;
   private final boolean mustBeSandboxed;
+  private final boolean proxyed;
```

Comment:
nit: Shouldn't this, and all other references, be "proxied"?

Also please add docstrings. What does it mean to be proxied?